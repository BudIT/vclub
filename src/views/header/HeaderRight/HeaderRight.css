.ul {
  list-style-type: none;
  margin: 0;
  padding: 0;
  display: flex;
  justify-content: flex-end;
  background-color: #FFD00B;
  flex: 1;
}

.tab {
  position: relative;

  color: #33363A;
  text-decoration: none;
  text-align: center;
  font-size: 22px;
  font-family: 'Rubik', helvetica, sans-serif;

  padding: 14px 16px;
  transition: 0.2s;
  display: inline-block;

  background-color: inherit;
  cursor: pointer;

  border-style: none;
  outline: none;

  border-right: 2px solid transparent;
}

.tab:hover {
  background-color: #FFAA00;
}

.tabDisabled {
  color: rgba(51, 54, 58, 0.5);
  cursor: default;
}

.tabDisabled:hover {
  background-color: transparent;
}

.errorMark {
  position: absolute;
  top: 25px;
  left: 25px;
  color: white;
  background-color: red;
  border-radius: 50%;
  font-size: 16px;
  font-family: 'Rubik', helvetica, sans-serif;
  text-align: center;
  width: 22px;
  height: 22px;
  line-height: 20px;
}

.dropdownContent {
  display: none;
  position: absolute;
  background-color: #FFD00B;
  min-width: 200px;
  box-shadow: 0px 8px 16px 0px rgba(0,0,0,0.2);
  right: 0px;
  z-index: 1;
}

.dropdown {}

.dropdownContent .tab:hover {
  background-color: #FFAA00;
  border-right: 2px solid #33363A;
}

.dropdown:hover .dropdownContent {
  display: flex;
  flex-direction: column;
}
