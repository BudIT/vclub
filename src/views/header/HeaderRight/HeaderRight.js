import React, { PropTypes } from 'react';

import compose from 'recompose/compose';
import { withHandlers } from 'recompose';

// actions
import { logOut } from 'vclub/redux/club/auth';
import { toggleMemberPanel } from 'vclub/redux/club/ui';

import style from './HeaderRight.css';

const enhance = compose(
  withHandlers({
    onToggleMemberPanel: props => () => {
      props.dispatch(toggleMemberPanel());
    },
    onLogOut: props => () => {
      props.dispatch(logOut());
    },
  })
);

function HeaderRight(props) {
  const {
    numberOfMembers,
    onLogOut, onToggleMemberPanel,
  } = props;

  return (
    <ul className={style.ul}>
      <li>
        <button
          className={style.tab}
          onClick={onToggleMemberPanel}
        >
          &#9977; {numberOfMembers}
        </button>
      </li>
      <li className={style.dropdown}>
        <button
          className={style.tab}
        >
          &#9776;
        </button>
        <div className={style.dropdownContent}>
          <button
            className={style.tab}
            onClick={onLogOut}
          >
            Log Out
          </button>
        </div>
      </li>
    </ul>
  );
}

HeaderRight.propTypes = {
  numberOfMembers: PropTypes.number.isRequired,
  onLogOut: PropTypes.func.isRequired,
  onToggleMemberPanel: PropTypes.func.isRequired,
};

HeaderRight.defaultProps = {
  numberOfMembers: 1,
};

export default enhance(HeaderRight);
