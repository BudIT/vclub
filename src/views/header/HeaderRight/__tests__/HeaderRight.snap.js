import React from 'react';
import renderer from 'react-test-renderer';
import HeaderRight from '../HeaderRight';

const numberOfMembers = 5;

test('<HeaderRight /> renders correctly', () => {
  const dispatchSpy = jest.fn();
  const rendered = renderer.create(
    <HeaderRight
      numberOfMembers={numberOfMembers}
      dispatch={dispatchSpy}
    />
  );
  expect(rendered.toJSON()).toMatchSnapshot();
});
