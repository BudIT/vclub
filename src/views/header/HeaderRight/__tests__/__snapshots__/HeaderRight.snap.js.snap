exports[`test <HeaderRight /> renders correctly 1`] = `
<ul
  className="ul">
  <li>
    <button
      className="tab"
      onClick={[Function]}>
      ⛹ 
      5
    </button>
  </li>
  <li
    className="dropdown">
    <button
      className="tab">
      ☰
    </button>
    <div
      className="dropdownContent">
      <button
        className="tab"
        onClick={[Function]}>
        Log Out
      </button>
    </div>
  </li>
</ul>
`;
